tests/test_input.py

```diff
@@ -12,16 +14,42 @@ def test_input_valid_working_dir_():
     # Assert path to DICOMs exists
     assert valid_filesystem_input.working_directory.is_dir()
 
+    valid_tcia_input = TCIAInput(series_instance_uid="")
+    # Assert path to DICOMs exists
+    assert os.path.isdir(valid_tcia_input.working_directory)
+
 
 def test_input_invalid_working_dir_():
-    invalid_test_input = WebInput(working_directory="NOT_VALID_PATH", data_url="")
+    invalid_test_input = WebInput(data_url="", working_directory="NOT_VALID_PATH")
     # Assert path to DICOMs does not exist
     assert not invalid_test_input.working_directory.is_dir()
 
     invalid_filesystem_input = FilesystemInput(working_directory="NOT_VALID_PATH")
     # Assert path to DICOMs does not exist
     assert not invalid_filesystem_input.working_directory.is_dir()
 
+    invalid_work_dir_tcia_input = TCIAInput(
+        series_instance_uid="1.3.6.1.4.1.14519.5.2.1.7695.4001.306204232344341694648035234440",
+        working_directory="NOT_VALID_PATH",
+    )
+    # Assert path to DICOMs does not exist
+    assert not invalid_work_dir_tcia_input.working_directory.is_dir()
+
+    invalid_series_uid_tcia_input = TCIAInput(series_instance_uid="NOT_VALID_SERIES_UID")
+    invalid_series_uid_tcia_input.fetch_data()
+    # Assert path to DICOMs does exist, but it contains no files
+    assert invalid_series_uid_tcia_input.working_directory.is_dir()
+    assert (
+        len(
+            [
+                name
+                for name in os.listdir(invalid_series_uid_tcia_input.working_directory)
+                if os.path.isfile(name)
+            ]
+        )
+        == 0
+    )
+
 
 def test_test_input():
```

Comment:
Sorry I know thats not part of this pull request, but the test_test_input function should be calling fetch_data on the test_input object, not the download_and_extract_zip_file function directly.